go_sequencer/common/account.go

```diff
@@ -1,67 +1,195 @@
 package common
 
 import (
+	"bytes"
 	"encoding/binary"
+	"errors"
 	"fmt"
 	"math/big"
+	"tokamak-sybil-resistance/common/nonce"
 
 	ethCommon "github.com/ethereum/go-ethereum/common"
 	"github.com/iden3/go-iden3-crypto/babyjub"
+	"github.com/iden3/go-iden3-crypto/poseidon"
+	cryptoUtils "github.com/iden3/go-iden3-crypto/utils"
 )
 
-// Account is a struct that gives information of the holdings of an address and
-// a specific token. Is the data structure that generates the Value stored in
+// Account is a struct that gives information of the holdings of an address.
+// Is the data structure that generates the Value stored in
 // the leaf of the MerkleTree
 type Account struct {
-	Idx      Idx                   `meddler:"idx"`
-	TokenID  TokenID               `meddler:"token_id"`
+	Idx      AccountIdx            `meddler:"idx"`
 	BatchNum BatchNum              `meddler:"batch_num"`
 	BJJ      babyjub.PublicKeyComp `meddler:"bjj"`
 	EthAddr  ethCommon.Address     `meddler:"eth_addr"`
 	Nonce    Nonce                 `meddler:"-"` // max of 40 bits used
 	Balance  *big.Int              `meddler:"-"` // max of 192 bits used
 }
 
-// Idx represents the account Index in the MerkleTree
-type Idx uint64
+// AccountIdx represents the account Index in the MerkleTree
+type AccountIdx uint64
 
 const (
-	// NLeafElems is the number of elements for a leaf
-	NLeafElems = 4
+	// NAccountLeafElems is the number of elements for a leaf in account tree
+	NAccountLeafElems = 4
```

Comment:
I think it is needed, because we have leaves seperated by `NAccountLeafElems`.

Of course, it can be hardcoded as like 128 rather than `32 * NAccountLeafElems`, but we need to make sure that those leaves are seperated by finite filed of 32 bytes, and the total number of elements of leaf is 4, that's the reason we have `NAccountLeafElems` here.

Additionally, It is more readable and understandable.